### PR TITLE
Feat playlist

### PR DESCRIPTION
**Pull request by VLeads**
## This PR will add functionality to the Playlist

Now the [Rapid TV](https://rapidtv-git-feat-playliast-vleads.vercel.app/) is working completely. You can do

- Log In
- Sign Up

### Watch a Video
- Watch a Video
- Clear All history at once

### Like a Video
- Check liked videos in liked page
- Remove video from liked videos

### Add video to watch later 
- Add video to watch later 
- Remove video from watch later

### Create Playlist
- Create a playlist
- Add video to playlist
- Delete Video from playlist
- Delete Specific playlist

#### Commit to check:👉  [Added playlist page with feature](https://github.com/VLeads/RapidTv/pull/5/commits/4d14e049e4790a4587496da17df95d3d0144538a)

Demo credentials

```
email: vishal@mail.com
password: Rapidstore1
```

### CODE REVIEW

**Review by JowelTisso — COMMENTED** [review]
* Too Big PR
* Try to keep your PR small

**Comment by VLeads** [comment]
> * Too Big PR
> * Try to keep your PR small

You don't need to check entire PR. I have mentioned the commit link to check. Only that commit need to be reviewed.

File: src/components/card/HistoryVideoCard.jsx [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    if (type === "fromhistory") {
+      deleteFromHistory(_id);
+    } else if (type === "fromwatchlater") {
+      deleteWatchLater(deleteWatchLaterVideoApi, _id);
+    } else if (type === "fromliked") {
+      deleteLike(deleteLikedVideoApi, _id);
+    } else if (type === "fromplaylist") {
+      deleteFromPlaylistHandler(extra._id, details._id, extra.title);
+    }
+  };
```
Comment (Developers-codz) [inline src/components/card/HistoryVideoCard.jsx:48]: this function can be extracted out, this way u can reuse it in all the other cards as well

File: src/components/card/card.module.css [hunk 2]
```diff
@@ -48,6 +50,85 @@
...
+
+.historyCardChannelName {
+    color: var(--COLOR-GREY-200);
+}
+
+.historyCardImageContainer img {
+    width: 20rem;
+    height: 100%;
+    object-fit: cover;
+    border-radius: 8px 0px 0px 8px;
```
Comment (Developers-codz) [inline src/components/card/card.module.css:75]: `0px` ?? `0` is better ?

File: src/components/sidebar/Sidebar.jsx [hunk 3]
```diff
@@ -14,12 +20,34 @@ export const Sidebar = () => {
...
+        <li>
+          <a href="https://twitter.com/vishalk01234" target="_blank">
+            <TwitterIcon />
+          </a>
+        </li>
+        <li>
+          <a href="mailto:leader.vishalkumar@gmail.com" target="_blank">
+            <MailIcon />
+          </a>
+        </li>
```
Comment (Developers-codz) [inline src/components/sidebar/Sidebar.jsx:49]: most of the jsx here is repeating, can make an array of object and loop over them ??

File: src/components/toast/toast.css [hunk 4]
```diff
@@ -0,0 +1,12 @@
+.toast-msg {
+    color: #f5f5f5
+}
+
+.success-toast {
+    background-color: #407ebd !important;
```
Comment (Developers-codz) [inline src/components/toast/toast.css:6]: avoid using `!important`  . It will override user stylesheets

**Review by Developers-codz — APPROVED** [review]
Looks good to merge 
Suggested some changes.
Also Please keep the PR small, It will be easier for us to review
One Observation in your site, Either fix the nav or make the aside sticky to top after scroll and video listing page cards have a lot of margin between them if u r using `justify-content` then do it `space-around` or provide less margin

File: src/components/card/HistoryVideoCard.jsx (hunk 1 above)
Comment (VLeads) [reply src/components/card/HistoryVideoCard.jsx:48]: Yes, i have extracted this function from playlist- context. And using wherever needed !!

File: src/components/card/card.module.css (hunk 2 above)
Comment (VLeads) [reply src/components/card/card.module.css:75]: Yups there should not be any unit with 0. Made the changes.

File: src/components/sidebar/Sidebar.jsx (hunk 3 above)
Comment (VLeads) [reply src/components/sidebar/Sidebar.jsx:49]: This is being only used at one place and for now they are only 4. So, I have implemented them directly on the very page without data.

File: src/components/toast/toast.css (hunk 4 above)
Comment (VLeads) [reply src/components/toast/toast.css:6]: I have imported toast from my component library, which comes with default color. Color was not working with specificity either. So, I used important tag there.

**Comment by VLeads** [comment]
> Looks good to merge Suggested some changes. Also Please keep the PR small, It will be easier for us to review One Observation in your site, Either fix the nav or make the aside sticky to top after scroll and video listing page cards have a lot of margin between them if u r using `justify-content` then do it `space-around` or provide less margin

Thanks Kajal for the detailed review. I have done some suggested changes and will consider some in new PRs.

**Review by Maianki — APPROVED** [review]

**Pull request merged**